stempy/reader.cpp

```diff
@@ -278,9 +278,22 @@ int extractSector(const std::string& fileName)
   return -1;
 }
 
-SectorStreamReader::SectorStreamReader(const vector<string>& files)
-  : m_files(files)
+SectorStreamReader::SectorStreamReader(const vector<string>& files,
+                                       uint8_t version)
+  : m_files(files), m_version(version)
 {
+  // Validate version
+  switch (this->m_version) {
```

Comment:
When using m_ for member variables it would be preferable to avoid using this-> as well